module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,16 @@
+#include "sort.hpp"
+
+#include <algorithm>
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> list) {
+    std::deque<std::string> deque;
+    std::copy(list.begin(), list.end(), std::back_inserter(deque));
+
+    std::sort(deque.begin(), deque.end(), [](const auto& lhs, const auto& rhs) {
+        if (lhs.size() == rhs.size()) {
+            return lhs < rhs;
+        }
+        return lhs < rhs;
```

Comment:
Hmm... without if it will work in the same way. This condition is not correct. And our tests sucks because they couldn’t detect such thing ;)

Reply:
Yes, it's my silly mistake. 13th line should be `return lhs.size() < rhs.size()`

Reply:
I added one more test that should prove that code in my newest commit works fine.